salt/transport/http.py

```diff
@@ -0,0 +1,363 @@
+# -*- coding: utf-8 -*-
+"""
+HTTP transport classes
+"""
+
+# Import Python Libs
+from __future__ import absolute_import, print_function, unicode_literals
+
+import json
+import base64
+import logging
+import socket
+import uuid
+
+# Import Tornado Libs
+import salt.ext.tornado
+import salt.ext.tornado.web
+import salt.ext.tornado.locks
+import salt.ext.tornado.gen
+import salt.ext.tornado.httpclient
+import salt.ext.tornado.httpserver
+import salt.transport.frame
+import salt.transport.abstract
+import salt.utils.msgpack
+from salt.transport.tcp import _set_tcp_keepalive
+from salt.ext import six
+
+log = logging.getLogger(__name__)
+
+if six.PY2:
+    import urllib
+    urlencode = urllib.urlencode
+else:
+    import urllib.parse
+    urlencode = urllib.parse.urlencode
+
+# pylint: disable=import-error,no-name-in-module
+if six.PY2:
+    import urlparse
+else:
+    import urllib.parse as urlparse
+# pylint: enable=import-error,no-name-in-module
+
+
+class AsyncHTTPReqChannel(salt.transport.abstract.AbstractAsyncReqChannel):
+    def __new__(cls, opts, **kwargs):
+        return super().__new__(cls, opts, **kwargs)
+
+    @classmethod
+    def __key(cls, opts, **kwargs):
+        return super().__key(cls, opts, **kwargs)
+
+    def start_channel(self, io_loop, **kwargs):
+        parse = urlparse.urlparse(self.opts["master_uri"])
+        master_host, master_port = parse.netloc.rsplit(":", 1)
```

Comment:
This doesn't work for IPv6 addresses. Use `parse.hostname` and `parse.port` instead